server/src/test/java/org/elasticsearch/snapshots/InternalSnapshotsInfoServiceTests.java

```diff
@@ -326,24 +385,28 @@ private ClusterState addUnassignedShards(final ClusterState currentState, String
             new IndexId(indexName, UUIDs.randomBase64UUID(random()))
         );
 
-        final Index index = metadata.get(indexName).getIndex();
-        final IndexRoutingTable.Builder indexRoutingTable = IndexRoutingTable.builder(index);
-        for (int primary = 0; primary < numberOfShards; primary++) {
-            final ShardId shardId = new ShardId(index, primary);
+        final IndexMetadata indexMetadata = metadata.get(indexName);
+        final Index index = indexMetadata.getIndex();
 
-            final IndexShardRoutingTable.Builder indexShards = new IndexShardRoutingTable.Builder(shardId);
-            indexShards.addShard(TestShardRouting.newShardRouting(shardId, null, true, ShardRoutingState.UNASSIGNED, recoverySource));
-            for (int replica = 0; replica < metadata.get(indexName).getNumberOfReplicas(); replica++) {
-                indexShards.addShard(TestShardRouting.newShardRouting(shardId, null, false, ShardRoutingState.UNASSIGNED,
-                    RecoverySource.PeerRecoverySource.INSTANCE));
-            }
-            indexRoutingTable.addIndexShard(indexShards.build());
+        final RoutingTable.Builder routingTable = RoutingTable.builder(currentState.routingTable());
```

Comment:
I made these changes so that the cluster state is better in line with a real cluster state containing an index to restore. Also, it allows to use an AllocationService to update the cluster state in order to initialize or start shards.